File: gnote.ts/src/__tests__/tags.test.ts
import { extractTagsFromText } from '../tags'

test('extractTagsFromText', () => {
  const result = extractTagsFromText({
    text: `
hello world
foo *bar*
link step
*baz* and 
		`,
    path: '/note',
    marker: '*',
  })
  expect(result).toEqual(
    `
bar\t/note\t/*bar*
baz\t/note\t/*baz*
	`.trim()
  )
})

File: gnote.go/src/commands/tags_test.go
package commands

import (
	"testing"
)

func TestExtractTagsFromText(t *testing.T) {
	text := `
		hello world
		foo *bar*
		link step
		*baz* and 
	`
	expected :=
		"bar\t/a.gnote\t/*bar*\n" +
			"baz\t/a.gnote\t/*baz*"
	result := extractTagsFromText(text, "/a.gnote", "*", createPattern("*"))
	if result != expected {
		t.Error("Failed")
	}
}
